contracts/MarginCalculator.sol

```diff
@@ -345,6 +368,128 @@ contract MarginCalculator is Ownable {
+            uint256,
+            uint256
+        )
+    {
+        // create LiquidationStatus struct to avoid stack too deep error
+        // will store isLiquidatable and the liquidation price
+        LiquidationStatus memory liquidationStatus = LiquidationStatus(false, 0);
+
+        // liquidation is only supported for naked margin vault
+        require(_vaultType == 1, "MarginCalculator: invalid vault type to liquidate");
+
+        // get vault details
+        VaultDetails memory vaultDetails = _getVaultDetails(_vault, _vaultType);
+
+        // can not liquidate vault that have no short position
+        if (!vaultDetails.hasShort) return (false, 0, 0);
+
+        // do not allow liquidating expired position
+        require(
+            now < vaultDetails.shortExpiryTimestamp,
+            "MarginCalculator: can not liquidate vault with expired short otoken"
+        );
+
+        // get otoken underlying asset price and timestamp from chainlink feed
+        (uint256 price, uint256 timestamp) = oracle.getChainlinkRoundData(
+            vaultDetails.shortUnderlyingAsset,
+            uint80(_roundId)
+        );
+
+        // check that price timestamp is after latest timestamp the vault was updated at
+        require(
+            timestamp >= _vaultLatestUpdate,
+            "MarginCalculator: auction timestamp should be post vault latest update"
+        );
+
+        // another struct to store some useful short otoken details, to avoid stack to deep error
+        ShortScaledDetails memory shortDetails = ShortScaledDetails({
+            shortAmount: FPI.fromScaledUint(_vault.shortAmounts[0], BASE),
+            shortStrike: FPI.fromScaledUint(vaultDetails.shortStrikePrice, BASE),
+            shortUnderlyingPrice: FPI.fromScaledUint(price, BASE)
+        });
+
+        // get product hash
+        bytes32 productHash = _getProductHash(
+            vaultDetails.shortUnderlyingAsset,
+            vaultDetails.shortStrikeAsset,
+            vaultDetails.shortCollateralAsset,
+            vaultDetails.isShortPut
+        );
+
+        // convert vault collateral to a fixed point (1e27) from collateral decimals
+        FPI.FixedPointInt memory collateralAmount = FPI.fromScaledUint(
+            _vault.collateralAmounts[0],
+            vaultDetails.collateralDecimals
+        );
+
+        // another scope to avoid stack too deep!
+        {
+            // get required collateral
+            FPI.FixedPointInt memory collateralRequired = _getNakedMarginRequired(
+                productHash,
+                shortDetails.shortAmount,
+                shortDetails.shortStrike,
+                shortDetails.shortUnderlyingPrice,
+                vaultDetails.shortExpiryTimestamp,
+                vaultDetails.isShortPut
+            );
+
+            // check if deposited collateral is greater or equal than needed collateral
+            liquidationStatus.isLiquidatable = collateralRequired.isGreaterThan(collateralAmount);
```

Comment:
better rename the `collateralAmount` variable to sometime more self explaining, like `depositedCollateral` so you don't need to explicitly say what you're comparing.

Reply:
will rename it to `availableCollateral`